gcovr/formats/__init__.py

```diff
@@ -84,6 +93,34 @@ def read_reports(options) -> CovData:
     else:
         covdata = GcovHandler(options).read_report()
 
+    if options.include:
+        for search_path in options.search_paths or [options.root]:
+            LOGGER.debug(f"Search for included files in {search_path}")
+            for fname in search_file(
+                lambda fname: any(f.match(fname) for f in options.include),
+                search_path,
+                exclude_dirs=options.gcov_exclude_dirs,
+            ):
+                # Return if the filename does not match the filter
+                # Return if the filename matches the exclude pattern
+                filtered, excluded = apply_filter_include_exclude(
+                    fname, options.filter, options.exclude
+                )
+
+                if filtered:
+                    LOGGER.debug(f"  Filtering coverage data for file {fname}")
```

Comment:
This is more of a debug question but it came up while I was trying to debug filters/excludes/etc. when using the tool. Could it be possible to know which pattern was matched when something was excluded/included/filtered? this would help when you mix and match several of these things together to know which ones had an effect.

Reply:
That's not possible for now because the filter function uses the `any` function of Python. This needs to be refactored to a `for` loop.

Reply:
I'll give it a try.

Reply:
Can you check the logging? I added a debug message to the filter class itself and refactored the filtering function where for every call some logging was always done afterwards.

Reply:
Works great and very informative now!